src/query/api/v1/handler/prom/read.go

```diff
@@ -249,6 +252,36 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// NB: this is a naive but lightweight method to extra a metric name from a PromQL query.
+// It returns an empty string if it fails to extract a metric name.
+// We don't want to parse the PromQL here because the extraction is not super important.
+func (h *readHandler) extractMetricName(query string) string {
```

Comment:
This only extracts the first metric name while query could be a combination of multiple metrics.

Reply:
I know. It should help the majority cases.